src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/health/QueueHealth.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.keyworker.services.health
+
+import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.model.*
+import com.amazonaws.services.sqs.model.QueueAttributeName.ApproximateNumberOfMessages
+import com.amazonaws.services.sqs.model.QueueAttributeName.ApproximateNumberOfMessagesNotVisible
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.Health.Builder
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.keyworker.services.health.QueueAttributes.*
+
+enum class DlqStatus(val description: String) {
+  UP("UP"),
+  NOT_ATTACHED("The queue does not have a dead letter queue attached"),
+  NOT_FOUND("The queue does not exist"),
+  NOT_AVAILABLE("The queue cannot be interrogated")
+}
+
+enum class QueueAttributes(val awsName: String, val healthName: String) {
+  MESSAGES_ON_QUEUE(ApproximateNumberOfMessages.toString(), "MessagesOnQueue"),
+  MESSAGES_IN_FLIGHT(ApproximateNumberOfMessagesNotVisible.toString(), "MessagesInFlight"),
+  MESSAGES_ON_DLQ(ApproximateNumberOfMessages.toString(), "MessagesOnDLQ")
+}
+
+@Component
+class QueueHealth(@Autowired @Qualifier("awsSqsClient") private val awsSqsClient: AmazonSQS,
+                  @Autowired @Qualifier("awsSqsDlqClient") private val awsSqsDlqClient: AmazonSQS,
+                  @Value("\${sqs.queue.name}") private val queueName: String,
+                  @Value("\${sqs.dlq.name}") private val dlqName: String) : HealthIndicator {
+
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun health(): Health {
+    val queueAttributes = try {
+      val url = awsSqsClient.getQueueUrl(queueName)
+      awsSqsClient.getQueueAttributes(getQueueAttributesRequest(url))
+    } catch (e: Exception) {
+      log.error("Unable to retrieve queue attributes for queue '{}' due to exception:", queueName, e)
+      return Builder().down().withException(e).build()
```

Comment:
Kotlin question; I assume the `return` breaks out of parent block? And without the return the result of the Builder() would be assigned to `val queueAttributes` ?

Reply:
agreed